packages/peregrine/lib/talons/OrderHistoryPage/useOrderRow.js

```diff
@@ -1,14 +1,72 @@
-import { useCallback, useState } from 'react';
+import { useCallback, useState, useMemo } from 'react';
+import { useQuery } from '@apollo/client';
+
+/**
+ * @function
+ *
+ * @param {Object} props
+ * @param {Array<Object>} props.items Collection of items in Order
+ * @param {OrderRowQueries} props.queries GraphQL queries for the Order Row Component
+ *
+ * @returns {OrderRowTalonProps}
+ */
+export const useOrderRow = props => {
+    const { items, queries } = props;
+    const { getProductThumbnailsQuery } = queries;
+
+    const skus = useMemo(() => {
+        return items.map(item => item.product_sku);
```

Comment:
Because apollo client caches by `toString` of query name + variables, and because these `skus` are used in the query below, you should sort them.

Reply:
So, even if the order of SKU's change, apollo cache hit will be a success?

Reply:
No, if the order of SKUs change, the cache will not "hit". The results will reference the same cached object after the network call but a network call will be made.

```
"query([sku1, sku3, sku2])": CACHED_DATA
"query([sku1, sku2, sku3])": CACHED_DATA
```

Reply:
Correct me if I am wrong. Because of sorting, the order of SKUs in the `skus` array passed to the query will stay the same even if their order changes in the `items` array. This will lead to cache hits every time after the first time.

Reply:
> Because of sorting, the order of SKUs in the skus array passed to the query will stay the same even if their order changes in the items array.

Yes, this is what I'm saying. You should sort the `skus` because array order is not guaranteed. The `items` array could be a different order.